src/Visualization/Python/PlotDatFile.py

```diff
@@ -0,0 +1,296 @@
+                            "of '--x-axis'.")
+
+        # Specify all plotting options
+        parser.add_argument('--x-bounds',
+                            required=False,
+                            nargs=2,
+                            help="The lower and upper bounds of the x-axis.")
+        parser.add_argument('--y-bounds',
+                            required=False,
+                            nargs=2,
+                            help="The lower and upper bounds of the y-axis.")
+        parser.add_argument('--x-label',
+                            required=False,
+                            help="The label on the x-axis. "
+                            "Default is the name of the x-axis entry.")
+        parser.add_argument('--linewidth',
+                            required=False,
+                            type=float,
+                            help="The width of the lines plotted.",
+                            default=plt.rcParams['lines.linewidth'])
+        parser.add_argument(
+            '--linestyles',
+            required=False,
+            nargs='+',
+            help="The styles of the lines plotted. Must be one or "
+            "one for each function.",
+            choices=['solid', 'dashed', 'dashdot', 'dotted'],
+            default=['solid'])
+        parser.add_argument('--legend-ncols',
+                            required=False,
+                            type=int,
+                            default=1,
+                            help="Number of columns for the legend")
+        parser.add_argument('--x-axis',
+                            required=False,
+                            help="If specified, make the range be the "
+                            "values is this field of the subfile. Defaults to "
+                            "the first column in the dat file.")
+        parser.add_argument('--y-logscale',
+                            required=False,
+                            action='store_true',
+                            help="If specified, set the y-axis to log scale.")
+
+        parser.add_argument('--title',
+                            required=False,
+                            help="Title on the graph.")
+        parser.add_argument('--fontsize',
+                            required=False,
+                            type=int,
+                            default=16,
+                            help="The font size on the plots")
+
+        parser.add_argument(
+            '--labels',
+            required=False,
+            type=json.loads,
+            help="R|A string containing a JSON dictionary to map some of\n"
+            "the functions to specified labels. An example of what\n"
+            "to pass to the argument (including the single quotes)\n"
+            "is:\n"
+            "'{\"Error(DivergenceCleaningField)\":\"$L_2(\\\\Phi)$\",\n"
+            "\"Error(RestMassDensity)\":\"$L_2(\\\\mathcal{E}(\\\\rho))$\"}'")
```

Comment:
Also make a note that if you are using LaTeX and you want to do something like `\mathcal{E}`, that you'll actually need to write `\\mathcal{E}`. I didn't do this, just wrote `--labels '{"Size on node 0 (MB)":"$\mathcal{E}$"}'` and got an error:

```
PlotDatFile.py: error: argument --labels: invalid loads value: '{"Size on node 0 (MB)":"$\\mathcal{E}$"}'
```

And then if I did `--labels '{"Size on node 0 (MB)":"$\beta$"}'` instead of `\\beta` I got a really long ugly traceback that ended in:

```
  File "/home/knelli/.local/lib/python2.7/site-packages/matplotlib/backends/backend_pdf.py", line 2153, in get_text_width_height_descent
    self.mathtext_parser.parse(s, 72, prop)
  File "/home/knelli/.local/lib/python2.7/site-packages/matplotlib/mathtext.py", line 3293, in parse
    box = self._parser.parse(s, font_output, fontsize, dpi)
  File "/home/knelli/.local/lib/python2.7/site-packages/matplotlib/mathtext.py", line 2521, in parse
    six.text_type(err)]))
ValueError: 
eta$
^
Expected end of text, found u'$'  (at char 0), (line:1, col:1)
```

Another idea is to just `try; catch` an exception when you're plotting and just print out something like "Unable to format labels. Make sure any LaTeX expressions have two `\\` instead of one"

Reply:
I'll add a note to the help string. Note that the help string already prints out with double slashes

Reply:
Yeah I saw that and thought it was a typo so I ignored it when testing....learned my lesson! The note you added is helpful.